### PR TITLE
StasticsHelper + Community Tab Performance Fix

### PR DESCRIPTION
**Pull request by W0lf4llo**
Hello! It's been a while since my last PR, I've been working on these fixes for a while as I was to lazy up until recently to actually fix these issues. The entire changelog is listed below!

- Total players online now show the exact amount of users playing said game, lets say there's 25 people online total, there's 5 playing LBP1, 10 playing LBP2, 5 playing Vita, and 5 playing LBP3. The game show you the amount of players playing the game you're currently on. So for LBP1 it'll show 5 total players online and so on!

- Team Picks total count now show the correct amount of levels that are able to be played on said game. I did a similar update to the total level count a while back as a test and seems to be working great! This update is similar, but it only affects the Team Pick total slot counts for game, this is very useful for vita players that want to actually see the amount of Team Picks that are on their server instance.

- Players in pod count, before when hovering over dive in it would show the total players online. Now it should display the amount of users that you're able to dive into, unfortunately I don't really know of a way to differentiate between PSN and RPCN users. If i do eventually find out I'll submit another PR fixing that issue!

- LBP3 Community Tab Performance Update, Yup, the LBP3 Community Tab should now be much smoother than before due to the "IEnumerable"'s changed to "IQueryable" Thanks to @sudokoko for this fix!

Special thanks to @Slendy, @daggintosh, & @sudokoko for making this all possible.

### CODE REVIEW

**Comment by sudokoko** [comment]
Hi there, thank you for your Project Lighthouse contribution. If you could sync your branch with upstream that would be awesome :)

File: ProjectLighthouse/Types/Entities/Level/Slot.cs [hunk 1]
```diff
@@ -163,7 +163,7 @@ public string[] LevelTags
 
     [XmlIgnore]
     public bool TeamPick { get; set; }
-
+
```
Comment (sudokoko) [inline ProjectLighthouse/Types/Entities/Level/Slot.cs:166]:
```suggestion
```
Remove unnecessary whitespace/tab

**Review by sudokoko — CHANGES_REQUESTED** [review]
1 minor nitpick but otherwise LGTM. Feel free to commit the change and I'll approve

**Review by sudokoko — APPROVED** [review]
LGTM

**Comment by daggintosh** [comment]
Insert nitpick

File: ProjectLighthouse.Servers.GameServer/Controllers/Slots/SlotsController.cs [hunk 2]
```diff
@@ -280,7 +280,7 @@ public async Task<IActionResult> TeamPickedSlots([FromQuery] int pageStart, [Fro
             .Take(Math.Min(pageSize, 30));
         string response = Enumerable.Aggregate(slots, string.Empty, (current, slot) => current + slot.Serialize(gameVersion));
         int start = pageStart + Math.Min(pageSize, ServerConfiguration.Instance.UserGeneratedContentLimits.EntitledSlots);
-        int total = await StatisticsHelper.TeamPickCount(this.database);
+        int total = await StatisticsHelper.TeamPickCountForGame(this.database, this.GetToken().GameVersion);
```
Comment (Slendy) [inline ProjectLighthouse.Servers.GameServer/Controllers/Slots/SlotsController.cs:283]:
```suggestion
        int total = await StatisticsHelper.TeamPickCountForGame(this.database, token);
```
Token already exists in the current context so there's no need to call GetToken again

**Review by Slendy — CHANGES_REQUESTED** [review]
Just one small change related to reusing a GetToken call

**Review by daggintosh — APPROVED** [review]
if this check fails you're fired

**Comment by daggintosh** [comment]
@Slendy you're fired.

File: ProjectLighthouse.Servers.GameServer/Controllers/Slots/SlotsController.cs [hunk 3]
```diff
@@ -280,7 +280,7 @@ public async Task<IActionResult> TeamPickedSlots([FromQuery] int pageStart, [Fro
             .Take(Math.Min(pageSize, 30));
         string response = Enumerable.Aggregate(slots, string.Empty, (current, slot) => current + slot.Serialize(gameVersion));
         int start = pageStart + Math.Min(pageSize, ServerConfiguration.Instance.UserGeneratedContentLimits.EntitledSlots);
-        int total = await StatisticsHelper.TeamPickCount(this.database);
+        int total = await StatisticsHelper.TeamPickCountForGame(this.database, token);
```
Comment (sudokoko) [inline ProjectLighthouse.Servers.GameServer/Controllers/Slots/SlotsController.cs:283]:
```suggestion
        int total = await StatisticsHelper.TeamPickCountForGame(this.database, token.GameVersion);
```

Slendy was wrong

**Review by Slendy — APPROVED** [review]

**Pull request merged**